src/servant/tt_001/F84.c

```diff
@@ -109,7 +109,104 @@ s32 UpdateEntityVelocityTowardsTarget(s32 unused, s32 targetX, s32 targetY) {
     return D_us_801735C0;
 }
 
-INCLUDE_ASM("servant/tt_001/nonmatchings/F84", func_us_80171284);
+Entity* func_us_80171284(
+    Entity* self) { // Assume self is also an Entity pointer
+    Entity* entity;
+    s32 i;
+    s32 index;
+    u32 matches = 0;
+#if !defined(VERSION_PSP)
+    s32 posDelta;
+#endif
+
+    // Hunt through these entities looking for ones that match all criteria.
+    // Call them a success and increment successes.
+    entity = &D_800762D8[0];
+    for (i = 0; i < 128; i++, entity++) {
+        D_us_801735C4[i] = 0;
+
+        // Very similar code to CheckAllEntitiesValid
+        if (!entity->entityId)
+            continue;
+        if (entity->hitboxState == 0)
+            continue;
+        if (entity->flags & FLAG_UNK_00200000)
+            continue;
+        if (entity->posX.i.hi < -16)
+            continue;
+        if (entity->posX.i.hi > 272)
+            continue;
+        if (entity->posY.i.hi > 240)
+            continue;
+        if (entity->posY.i.hi < 0)
+            continue;
+            // Differs from here
+#if defined(VERSION_PSP)
+        if (entity->hitboxState & 8 &&
+            D_us_80170510[D_us_80173810.level / 10][2] == 0)
+            continue;
+        if (abs(self->posX.i.hi - entity->posX.i.hi) >= 49)
+            continue;
+#else
+        if (entity->hitboxState & 8 &&
+            D_us_80170510[D_us_80173810.level / 10][0] == 0)
+            continue;
+        posDelta = self->posX.i.hi - entity->posX.i.hi;
+        if (posDelta < 0) {
+            posDelta *= -1;
+        }
+        if (posDelta >= 49)
+            continue;
+#endif
+
+#if defined(VERSION_PSP)
+        if (abs(self->posY.i.hi - entity->posY.i.hi) >= 49)
```

Comment:
My suspicion is that an abs macro is being used on both systems like
```
#define ABS(x) ((x) < 0 ? -(x) : (x))
```
but I wasn't able to get a match

Reply:
Have incorporate this for PSX to clean up the code a little as discussed